keps/sig-network/2681-pod-host-ip/README.md

```diff
@@ -0,0 +1,534 @@
+retains the existing (singular) "HostIP" field in the core V1 version of the
+PodStatus V1 core API and adds a new array of structures that store host IPs
+along with associated metadata for that IP.
+
+``` go
+    // HostIP represents the IP address of a host.
+    // IP address information. Each entry includes:
+    //    IP: An IP address allocated to the host.
+    type HostIP struct {
+      // ip is an IP address (IPv4 or IPv6) assigned to the host
+      IP string
+    }
+
+    // HostIPs holds the IP addresses allocated to the host. If this field is specified, the 0th entry must
+    // match the hostIP field. This list is empty if no IPs have been allocated yet.
+    // +optional
+    HostIPs []HostIP
+```
+
+### PodStatus Internal Representation
+
+PodStatus internally indicates that the original use of HostIP will remain unchanged,
+and HostIPs is added for subsequent use
+
+``` go
+    // HostIP address information for entries in the (plural) HostIPs field.
+    // Each entry includes:
+    //    IP: An IP address allocated to the host.
+    type HostIP struct {
+      // ip is an IP address (IPv4 or IPv6) assigned to the host
+      IP string `json:"ip,omitempty" protobuf:"bytes,1,opt,name=ip"`
+    }
+
+    // HostIPs holds the IP addresses allocated to the host. If this field is specified, the 0th entry must
+    // match the hostIP field. This list is empty if no IPs have been allocated yet.
+    // +optional
+    // +patchStrategy=merge
+    // +patchMergeKey=ip
+    HostIPs []HostIP `json:"hostIPs,omitempty" protobuf:"bytes,14,rep,name=hostIPs" patchStrategy:"merge" patchMergeKey:"ip"`
+```
+
+### Maintaining Compatible Interworking between Old and New Clients
+
+[See the making a singular field plural](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#making-a-singular-field-plural)
+
+### Container Environment Variables
+
+The Downward API [status.hostIP](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#capabilities-of-the-downward-api)
+will preserve the existing single IP address, and will be set to the default IP
+for each pod. A new environmental variable named `status.hostIPs` will contain a
+space-separated list of IP addresses. The new pod API will have a slice of
+structures for the additional IP addresses. Kubelet will translate the pod
+structures and return hostIPs as a comma-delimited string.
+
+Here is an example of how to define a pluralized `MY_HOST_IPS` environmental
+variable in a pod definition yaml file:
+
+``` yaml
+- name: MY_HOST_IPS
+  valueFrom:
+    fieldRef:
+      fieldPath: status.hostIPs
+```
+
+This definition will cause an environmental variable setting in the pod similar
+to the following:
+
+```
+# PodHostIPs FeatureGate is enabled
+MY_HOST_IPS=fd00:10:20:0:3::3,10.20.3.3
+
+# PodHostIPs FeatureGate is disabled
+MY_HOST_IPS=fd00:10:20:0:3::3
```

Comment:
If the gate is off, the "hostIPs" should do nothing at all - it should act as if it doesn't exist (including a validation error if needed)